src/context.ts

```diff
@@ -1,10 +1,12 @@
 import { ITranslator, nullTranslator } from '@jupyterlab/translation';
 import React from 'react';
 
-export type Logger = (eventName: string) => void;
-export const LogContext = React.createContext<Logger>((eventName: string) => {
-  /*noop*/
-});
+export type Logger = (eventName: string, error?: Error) => void;
```

Comment:
Making error as `Error` type is too specific to HTTP errors. Though the intent here is to primarily capture the HTTP errors, having something simpler can accomplish this. 

```suggestion
export type Logger = (eventName: string, eventDetail?: string) => void;
```